bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/GarbageCollectorThread.java

```diff
@@ -653,6 +653,8 @@ public synchronized void shutdown() throws InterruptedException {
         }
         LOG.info("Shutting down GarbageCollectorThread");
 
+        throttler.cancelledAcquire();
+        compactor.throttler.cancelledAcquire();
```

Comment:
#2963 introduces another limiter in the GC thread. When the current limit is set very small, it will also cause the GC thread to hang for a long time, so the current limit interrupt logic Move to Throttler.

In this way, where necessary, we can use the `Throttler.cancelledAcquire` method to interrupt the long-term thread sleep caused by current limiting.